lms/services/application_instance.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from sqlalchemy import or_
+
+from lms.models import ApplicationInstance
+
+
+class ApplicationInstanceService:
```

Comment:
There's `ApplicationInstanceGetter` service but that seems to be very focused on the application instance currently in use so I don't think these fit there

Reply:
Now they share the code for the `.get`